google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/Exceptions.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigquery.storage.v1;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.gax.grpc.GrpcStatusCode;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import javax.annotation.Nullable;
+
+/** Exceptions for Storage Client Libraries. */
+public final class Exceptions {
+  /** Main Storage Exception. Might contain map of streams to errors for that stream. */
+  public static class StorageException extends RuntimeException {
+
+    private final ImmutableMap<String, GrpcStatusCode> errors;
+    private final String streamName;
+
+    private StorageException() {
+      this(null, null, null, ImmutableMap.of());
+    }
+
+    private StorageException(
+        @Nullable String message,
+        @Nullable Throwable cause,
+        @Nullable String streamName,
+        ImmutableMap<String, GrpcStatusCode> errors) {
+      super(message, cause);
+      checkArgument(errors != null, "errors must be non null");
+      this.streamName = streamName;
+      this.errors = errors;
+    }
+
+    public ImmutableMap<String, GrpcStatusCode> getErrors() {
+      return errors;
+    }
+
+    public String getStreamName() {
+      return streamName;
+    }
+  }
+
+  /** Stream has already been finalized. */
+  public static final class StreamFinalizedException extends StorageException {
+    protected StreamFinalizedException(String name, String message, Throwable cause) {
+      super(message, cause, name, null);
```

Comment:
This will fail the null check in the constructor, will need to provide an empty collection instead.
```suggestion
      super(message, cause, name, ImmutableMap.of());
```